exotations/task_maps/task_map/include/task_map/ConvexHull.h

```diff
@@ -0,0 +1,80 @@
+#ifndef CONVEXHULL_H
+#define CONVEXHULL_H
+
+#include "Eigen/Dense"
+#include "exotica/Tools.h"
+
+namespace exotica
+{
+double lineDist2D(Eigen::VectorXdRefConst p1, Eigen::VectorXdRefConst p2, Eigen::VectorXdRefConst p)
+{
+    return (p(1) - p1(1)) * (p2(0) - p1(0)) -
+           (p2(1) - p1(1)) * (p(0) - p1(0));
+}
+
```

Comment:
Hmm, is this function working as it is supposed to, @VladimirIvan ?

```python
>>> def lineDist2D(p1, p2, p):
...     return (p[1] - p1[1]) * (p2[0] - p1[0]) - (p2[1] - p1[1]) * (p[0] - p1[0])
... 
>>> lineDist2D([0, 0], [0, 0], [1, 1])
0
>>> lineDist2D([0, 0], [2, 0], [1, 1])
2
>>> lineDist2D([0, 0], [-2, 0], [1, 1])
-2
>>> lineDist2D([-2, 0], [0, 0], [1, 1])
2
>>> lineDist2D([0, 0], [1, 0], [1, 1])
1
```

Reply:
Oh boy, you're right. It's supposed to be:
```
def lineDist2D(p1, p2, p):
  return ((p[1] - p1[1]) * (p2[0] - p1[0]) - (p2[1] - p1[1]) * (p[0] - p1[0]))/math.sqrt( (p2[0]-p1[0])**2 + (p2[1]-p1[1])**2 )
```

Classic vlad

Reply:
```python
def lineDist2D(p1, p2, p):
    num = (p[1] - p1[1]) * (p2[0] - p1[0]) - (p2[1] - p1[1]) * (p[0] - p1[0])
    denom = math.sqrt((p2[0] - p1[0]) ** 2 + (p2[1] - p1[1]) ** 2)
    return num / denom
```

This does not handle some corner cases. For example:
```python
>>> lineDist2D([0, 0], [0, 0], [1, 1])Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 4, in lineDist2D
ZeroDivisionError: float division by zero
```

Also, the lack of documentation or a more descriptive name makes me a bit confused. Is the function supposed to compute the shortest distance from a given point to any point in the line? Or does it compute the *y*-distance? Here's why I ask:
```python
>>> lineDist2D([0, 0], [1, 0], [2, 0])
0.0
```

Reply:
What I was aiming for was perpendicular oriented distance be tween a point and a line. It is actually not defined when the line points are coincident. The division by zero can be guarded and return 0.

Perhaps you got it confused with a distance to a line segment. That can't be signed and it is not what's needed for the quickhull algorithm.

Although, I have just realized that you actually don't need to divide by `||p2-p1||` for the quickhull to work. It does need a new name though. I believe it's called a [determinant in that case](http://mathworld.wolfram.com/CrossProduct.html).

Reply:
Something like this, right?
![img](https://i.imgur.com/fnQnjp9.png)